File: stepik-java-api/src/main/java/org/stepik/api/queries/units/StepikUnitsGetQuery.java
package org.stepik.api.queries.units;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.units.Units;
import org.stepik.api.queries.StepikAbstractGetQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikUnitsGetQuery extends StepikAbstractGetQuery<StepikUnitsGetQuery, Units> {
    public StepikUnitsGetQuery(@NotNull StepikAbstractAction stepikAction) {
        super(stepikAction, Units.class);
    }

    @NotNull
    public StepikUnitsGetQuery page(int value) {
        addParam("page", value);
        return this;
    }

    @NotNull
    public StepikUnitsGetQuery lesson(long value) {
        addParam("lesson", value);
        return this;
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.UNITS;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/units/StepikUnitsGetQuery.kt
package org.stepik.api.queries.units

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.units.Units
import org.stepik.api.queries.StepikAbstractGetQuery

class StepikUnitsGetQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractGetQuery<StepikUnitsGetQuery, Units>(stepikAction, Units::class.java) {
    
    fun page(value: Int): StepikUnitsGetQuery {
        addParam("page", value)
        return this
    }
    
    fun lesson(value: Long): StepikUnitsGetQuery {
        addParam("lesson", value)
        return this
    }
    
    override fun getUrl(): String {
        return "${stepikAction.stepikApiClient.host}/api/units"
    }
}
